### PR TITLE
feat(blog): add skeleton loader for blogs

### LINKED ISSUE #75: [Enhancement] Add Skeleton Screen Effect while Loading Blog
The current effect of loading blog may let user confuse what will load, we can add the skeleton screen effect to improve the user experience 

### Current Screenshots
| | |
|-|-|
| image | image |

### Reference 

- https://mui.com/material-ui/react-skeleton/
- https://www.npmjs.com/package/react-loading-skeletons
- https://www.smashingmagazine.com/2020/04/skeleton-screens-react/
- https://blog.logrocket.com/build-skeleton-screen-css/

### PR DESCRIPTION
**Pull request by irfen-mi**

### CODE REVIEW

**Comment by 1chooo** [comment]
@mu-irfan is there any demo or screenshots? In the preview I don't find the effects of skeleton frame. 

Is it the result?
Screenshot 2024-09-22 at 3 15 09 PM

**Comment by irfen-mi** [comment]
@1chooo 
yeh checkout this

https://github.com/user-attachments/assets/f2d272e6-26e0-4480-b6e9-d1c4730c4bfc

**Comment by 1chooo** [comment]
> @1chooo yeh checkout this
> 
>  Screen.Recording.2024-09-22.at.10.07.57.AM.mp4

That's awesome! However, I wasn't able to find it in the Vercel preview.

File: apps/web/src/app/blog/page.tsx [hunk 1]
```diff
@@ -49,54 +63,77 @@ export default function BlogPage({ searchParams }: { searchParams: { tag?: strin
...
-                    <span className="dot"></span>
-                    <time dateTime={post.metadata.publishedAt}>
-                      {new Date(post.metadata.publishedAt).toLocaleDateString(
-                        'en-us', { month: 'long', day: 'numeric', year: 'numeric' }
-                      )}
-                    </time>
+          {isLoading
+            ? Array.from({ length: POSTS_PER_PAGE }).map((_, index) => (
+                <li key={index} className="blog-post-item skeleton">
+                  <Skeleton height={200} className="blog-banner-skeleton" />
```
Comment (1chooo) [inline apps/web/src/app/blog/page.tsx:70]: Please use `SkeletonTheme` with baseColor, `#202020`, and highlightColor, `#444`

```tsx
<SkeletonTheme baseColor="#202020" highlightColor="#444">
  <p>
    <Skeleton count={3} />
  </p>
</SkeletonTheme>
```

File: apps/web/src/app/blog/page.tsx [hunk 2]
```diff
@@ -49,54 +63,77 @@ export default function BlogPage({ searchParams }: { searchParams: { tag?: strin
...
+                  <div className="blog-content-skeleton">
+                    <Skeleton width={100} className="blog-category-skeleton" />
```
Comment (1chooo) [inline apps/web/src/app/blog/page.tsx]: remove `blog-category-skeleton`.

File: apps/web/src/app/blog/page.tsx [hunk 3]
```diff
@@ -49,54 +63,77 @@ export default function BlogPage({ searchParams }: { searchParams: { tag?: strin
...
+                    <Skeleton width={180} className="blog-meta-skeleton" />
```
Comment (1chooo) [inline apps/web/src/app/blog/page.tsx]: remove `blog-meta-skeleton`.

**Review by 1chooo — CHANGES_REQUESTED** [review]

**Comment by irfen-mi** [comment]
Well, this skeleton loader will be display whenever blogs takes time to load, in vercel blogs maybe load quickly that's why, but if you want to display the skeleton loader compulsory whether blogs is load or not then i can put settimeout for a second or 2 maybe.

**Comment by 1chooo** [comment]
> Well, this skeleton loader will be display whenever blogs takes time to load, in vercel blogs maybe load quickly that's why, but if you want to display the skeleton loader compulsory whether blogs is load or not then i can put settimeout for a second or 2 maybe.

Yeah! I think you are right; I think we don't need to set the timeout anymore! There is still some changes need your help! Please check the review and show the demo of the changes.

**Comment by irfen-mi** [comment]
@1chooo Alright okay, i will work on your above requirements, then will push the code.

**Comment by 1chooo** [comment]
> @1chooo Alright okay, i will work on your above requirements, then will push the code.

Also, you can add your information to the document `apps/docs/pages/thanks.mdx` blog part 🥳

**Comment by irfen-mi** [comment]
@1chooo okay i will add for sure!

**Comment by 1chooo** [comment]
Hi @mu-irfan May I ask approximately how much time it will take? I have other tasks that are gated by this PR.

**Comment by irfen-mi** [comment]
@1chooo 
i have push the code, please review it, also here is demo attached

https://github.com/user-attachments/assets/06a91325-bc10-4aca-a345-0d9ec0cdec77

File: apps/docs/pages/thanks.mdx
Comment (1chooo) [inline apps/docs/pages/thanks.mdx:1]: Maybe you can also add your contact info! Really appreciate

**Review by 1chooo — APPROVED** [review]
LGTM

File: apps/docs/pages/thanks.mdx
Comment (irfen-mi) [reply apps/docs/pages/thanks.mdx:1]: i am adding it here: 
Whatsapp no: +92 3422022174

**Pull request merged**